sample_tracking/app/import-samples.tsx

```diff
@@ -0,0 +1,290 @@
+import { firebaseConfig } from './firebase_config';
+import { getFirestore, getDoc, doc, writeBatch } from "firebase/firestore";
+import { useTranslation } from 'react-i18next';
+import './i18n/config';
+import Papa from 'papaparse';
+import { type Sample, type UserData, validateImportedEntry, getRanHex } from './utils';
+import { ExportToCsv } from 'export-to-csv';
+
+/**
+ * Component to handle importing samples from a csv file. 
+ * 
+ * Checks if the data being imported has the required data and the result valuesa are in the correct format.
+ * If format is correct, data is uploaded and a success bar is shown on the main samples page. 
+ * If format is incorrect, error bar is shown on main samples page and gives option to download the same csv
+ * file with an extra column with the errors written out to tell the user how to fix their file to be properly imported.
+ * 
+ * CSV file should be a list of single result values which are linked together with the same 'code' value. These result
+ * values are joined together into a single "Sample" to be uploaded to the corresponding collection in the firestore database. 
+ */
+export default function ImportSamples() {
+    const [userData, setUserData] = useState(null as UserData | null)
+
+    const [errorSamples, setErrorSamples] = useState(null as Sample[] | null);
+
+    const app = initializeApp(firebaseConfig);
+    const router = useRouter();
+    const auth = getAuth();
+    const db = getFirestore();
+    const { t } = useTranslation();
+
+    const csvOptions = {
+        fieldSeparator: ',',
+        quoteStrings: '"',
+        decimalSeparator: '.',
+        showLabels: true,
+        useBom: true,
+        useKeysAsHeaders: true,
+    };
+    const csvExporter = new ExportToCsv(csvOptions);
+
+
+    const originValues = {
+        unknown: 'unknown',
+        known: 'trusted',
+        uncertain: 'untrusted'
+    }
+
+    function onImportSuccessBar() {
+        return (<div className="import-success-status-wrapper success-background-color">
+            <div className='import-status-icon-wrapper'>
+                <div className='import-status-icon'>
+                    <span className="material-symbols-outlined import-status-icon icon-color-green">
+                        check_circle
+                    </span>
+                </div>
+            </div>
+            <div className='import-status-text-wrapper'>
+                <div className='import-status-text text-color-green'>
+                    Successfully imported file
+                </div>
+            </div>
+            <div className='import-status-actions-wrapper'>
+                <div className='import-status-actions'>
+                    <div onClick={handleCloseBarClick} className="import-success-status-button pointer">
+                        <div className='import-status-button-slate-layer'>
+                            <div className='import-status-button-text'>
+                                Great!
```

Comment:
is Great! supposed to be here?

Reply:
Haha its in the mocks 
https://www.figma.com/file/t9y5EMZSeoXhLIwTuFxs9u/TimberID-spec?node-id=720%3A286209&mode=dev